conans/model/build_info.py

```diff
@@ -167,8 +177,11 @@ class CppInfo(_CppInfo):
     to build properly.
     Defined in user CONANFILE, directories are relative at user definition time
     """
-    def __init__(self, root_folder):
+
+    def __init__(self, ref_name, root_folder):
         super(CppInfo, self).__init__()
+        self._ref_name = ref_name
```

Comment:
Why the 2 variables ``_ref_name`` and ``_name``? As private I don't see them being assigned or being different anywhere. Am i missing something?

Reply:
`_ref_name` stores the same value forever, it is the name of the package (canonical, from the reference). 

`_name` is initialized with this value, but the user can modify it with `self.cpp_info.name = "Xyz"`